src/types/events/identify.rs

```diff
@@ -1,22 +1,153 @@
+    // I wonder if these are interchangable..
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub intents: Option<i32>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub capabilities: Option<i32>,
+}
+
+impl Default for GatewayIdentifyPayload {
+    fn default() -> Self {
+        Self::common()
+    }
+}
+
+impl GatewayIdentifyPayload {
+    /// Uses the most common, 25% data along with client capabilities
+    /// 
+    /// Basically pretends to be an official client on windows 10, with chrome 113.0.0.0
+    pub fn common() -> Self {
+        Self { token: "".to_string(), properties: GatewayIdentifyConnectionProps::default(), compress: Some(false), large_threshold: None, shard: None, presence: None, intents: None, capabilities: Some(8189)  }
+    }
+}
+
+impl GatewayIdentifyPayload {
+    /// Creates an identify payload with the same default capabilities as the official client
+    pub fn default_w_client_capabilities() -> Self {
+        let mut def = Self::default();
+        def.capabilities = Some(8189); // Default capabilities for a client
+        def
+    }
+
+    /// Creates an identify payload with all possible capabilities
+    pub fn default_w_all_capabilities() -> Self {
+        let mut def = Self::default();
+        def.capabilities = Some(i32::MAX); // Since discord uses bitwise for capabilities, this has almost every bit as 1, so all capabilities
+        def
+    }
 }
 
 impl WebSocketEvent for GatewayIdentifyPayload {}
 
-#[derive(Debug, Deserialize, Serialize, Default)]
+#[derive(Debug, Deserialize, Serialize)]
 pub struct GatewayIdentifyConnectionProps {
+    /// Almost always sent
+    /// 
+    /// ex: "Linux", "Windows", "Mac OS X"
+    /// 
+    /// ex (mobile): "Windows Mobile", "iOS", "Android", "BlackBerry"
     pub os: String,
+    /// Almost always sent
+    /// 
+    /// ex: "Firefox", "Chrome", "Opera Mini", "Opera", "Blackberry", "Facebook Mobile", "Chrome iOS", "Mobile Safari", "Safari", "Android Chrome", "Android Mobile", "Edge", "Konqueror", "Internet Explorer", "Mozilla", "Discord Client"
     pub browser: String,
+    /// Sometimes not sent, acceptable to be ""
+    /// 
+    /// Speculation:
+    /// Only sent for mobile devices
+    /// 
+    /// ex: "BlackBerry", "Windows Phone", "Android", "iPhone", "iPad", ""
     pub device: String,
+    /// Almost always sent, most commonly en-US
+    /// 
+    /// ex: "en-US"
+    pub system_locale: String,
+    /// Almost always sent
+    /// 
+    /// ex: any user agent, most common is "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/113.0.0.0 Safari/537.36"
+    pub browser_user_agent: String,
+    /// Almost always sent
+    /// 
+    /// ex: "113.0.0.0"
+    pub browser_version: String,
+    /// Sometimes not sent, acceptable to be ""
+    /// 
+    /// ex: "10" (For os = "Windows")
+    pub os_version: String,
+    /// Sometimes not sent, acceptable to be ""
```

Comment:
```suggestion
    /// Sometimes not sent, acceptable to be ""
```
If it is acceptable for these attributes to be empty strings, but it is not acceptable for them to be non-existent, we should probably look at https://docs.rs/serde_with/latest/serde_with/struct.NoneAsEmptyString.html

Reply:
...to properly represent that filling out this attribute is `Option<>`al